### PR TITLE
Fixes device scheduling unit tests

### PR DESCRIPTION
**Pull request by preetapan**
Also changes the logic for score when there is more than one task
requesting a device. Since inter task affinities are already normalized,
we take the average of the scores across tasks.

### CODE REVIEW

File: scheduler/rank.go [hunk 1]
```diff
@@ -352,8 +350,8 @@ OUTER:
 		iter.ctx.Metrics().ScoreNode(option.Node, "binpack", normalizedFit)
 
 		// Score the device affinity
-		if totalDeviceAffinityWeight != 0 {
-			deviceAffinityScore /= totalDeviceAffinityWeight
+		if devicesWithAffinities != 0 {
```
Comment (dadgar) [inline scheduler/rank.go]: != 0.0 and drop the float cast below

File: scheduler/device.go [hunk 2]
```diff
@@ -76,7 +78,7 @@ func (d *deviceAllocator) AssignDevice(ask *structs.RequestedDevice) (out *struc
 					continue
 				}
 
-				totalWeight += a.Weight
```
Comment (dadgar) [inline scheduler/device.go]: This should return choice score unnormalized

File: scheduler/rank.go [hunk 3]
```diff
@@ -196,7 +196,7 @@ OUTER:
 		devAllocator.AddAllocs(proposed)
 
 		// Track the affinities of the devices
-		totalDeviceAffinityWeight := 0.0
+		devicesWithAffinities := 0.0
```
Comment (dadgar) [inline scheduler/rank.go]: the final math should be the (sum(matching weight returned by device allocator))/sum(all device weights)

File: scheduler/device_test.go [hunk 4]
```diff
@@ -302,7 +302,6 @@ func TestDeviceAllocator_Allocate_Affinities(t *testing.T) {
 				},
 			},
 			ExpectedDevice: nvidia0,
-			ZeroScore:      true,
 		},
```
Comment (preetapan) [inline scheduler/device_test.go]: score will be -0.2 here not zero.

File: scheduler/device.go [hunk 5]
```diff
@@ -76,14 +78,15 @@ func (d *deviceAllocator) AssignDevice(ask *structs.RequestedDevice) (out *struc
...
+				totalWeight += math.Abs(a.Weight)
 
 				// Check if satisfied
 				if !checkAttributeAffinity(d.ctx, a.Operand, lVal, rVal) {
 					continue
 				}
 				choiceScore += a.Weight
 			}
+			matchedWeights += choiceScore
```
Comment (dadgar) [inline scheduler/device.go]: Hmm this is over accounting. Shouldn't this be after: https://github.com/hashicorp/nomad/pull/4855/files#diff-2ffd1ab21e7ba9159b1cd9f8919596edR127

File: scheduler/device.go [hunk 6]
```diff
@@ -39,7 +41,7 @@ func (d *deviceAllocator) AssignDevice(ask *structs.RequestedDevice) (out *struc
 	// Hold the current best offer
 	var offer *structs.AllocatedDeviceResource
 	var offerScore float64
-
+	var matchedWeights float64
```
Comment (dadgar) [inline scheduler/device.go]: Add a blank line below

**Pull request merged**